src/ripple/server/impl/BaseWSPeer.h

```diff
@@ -394,10 +462,18 @@ on_timer(error_code ec)
 {
     if(ec == boost::asio::error::operation_aborted)
         return;
-    if(! ec)
+
+    if (! ec && timer_action_ != 0)
```

Comment:
I don't know how `timer_action_` could be 2 (we do a _really_ long write?), but it seems like we're really checking for an unanswered ping.
What do you think about changing this to:
```
if (! ec && timer_action_ == 1)
```